### DISCUSSION TITLE
about division truncation in godot (math problem)

### DISCUSSION THREAD

**Discussion by 2439905184** [godotengine/godot-proposals#7284] [Engine Core]
### Describe the project you are working on

in godot 3.5.2
in music game , i need to calcuate  Timing of each beat
for example
music bpm:120 (120 times pre minute)
Timing of each beat = 60.0/120.0=0.5 (math calcuate)
but when cal in godot , this always return 0
how to solve this problem , when I need small float / big float , then get the 0.x like value?
this is a math problem

### Describe the problem or limitation you are having in your project

in math calcuate

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

give some math helps about things like 60.0 / 120.0 = 0.5
but 60.0 /120.0 get 0.0 in godot
I need this 0.5
how to get it.
if engine coder can't resolve this , how to use * math and other rules of arithmetic to get this 0.5?
MicoSoft's windows's calcuate can return true value , how to get this in godot?
![image](https://github.com/godotengine/godot-proposals/assets/29478722/f0d6ac01-2a35-41d0-a71f-9a892d53379a)

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

no

### If this enhancement will not be used often, can it be worked around with a few lines of script?

still in think （another math calcuate to get this <small float>/<big float> get a true float in godot）

### Is there a reason why this should be core and not an add-on in the asset library?

nothing

**Comment by AThousandShips** [comment]
> Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams
no

You need to fill this in, otherwise you will need to open a discussion, proposals require that some basic idea of how to implement it exists

Will convert to discussion as this lacks required information

**Comment by Calinou** [comment]
>  MicoSoft's windows's calcuate can return true value , how to get this in godot?

Use a floating-point number as one of the operands: `3.0/4`

**Comment by YuriSizov** [comment]
Hey there! I think you have an issue in your code somewhere. The only way for `60.0 / 120.0` to return `0` is for it to actually be `60 / 120` instead.

To explain, in Godot there is a difference between an integer division and a floating point number division. If you divide two integers you get an integer. If you divide an integer and a float, or two floats, you get a float. You seem to operate under the assumption that in your code you have floating point numbers. But that doesn't seem to be true.

You can ensure that you operate on floats by casting your values. E.g. `float(x) / float(y)`.

**Comment by AThousandShips** [comment]
See [documentation](https://docs.godotengine.org/en/stable/tutorials/scripting/gdscript/gdscript_basics.html#operators)
